app/controllers/spam2_controller.rb

```diff
@@ -0,0 +1,132 @@
+class Spam2Controller < ApplicationController
+  before_action :require_user, only: %i(spam spam_revisions mark_comment_spam publish_comment spam_comments)
+
+  def _spam
+    if logged_in_as(%w(moderator admin))
+      @nodes = Node.paginate(page: params[:page])
+      @nodes = if params[:type] == 'wiki'
```

Comment:
It might be a bit more efficient and clearer here to first get the specific nodes and then paginate them.

Reply:
Done, thanks!